### ISSUE TITLE
File creation behavior

### ISSUE THREAD

**Issue by Xemnarth** [BiglySoftware/BiglyBT#3219]
Unsure if this qualifies as an issue but when compared to other mainstream clients (such as qBittorent), as soon as a transfer is started all files within are created. Tried tweaking the Options under Files to no avail. Files are ALWAYS written even with 0 bytes...

**Comment by Xemnarth** [comment]
hmmm, must be an issue with my system/config then...

**Comment by parg** [comment]
Must be because I do this every day.

**Comment by parg** [comment]
If you right-click on one of these 0 length files (in BiglyBT) and select "set priority->delete" does it get deleted on disk?

**Comment by Xemnarth** [comment]
> edge-pieces

files containing edge-pieces are still created even with "set priority->delete" or with them being unselected/unmarked before being added to the transfer list.

**Comment by parg** [comment]
correct

**Comment by Xemnarth** [comment]
can this file creation behavior be changed so that nothing is created unless at least more than 0 B is downloaded?

**Comment by Xemnarth** [comment]
yes, but I mean even with a torrent containing a single file of which nothing is downloaded (0 byte), the client still creates it as soon as the transfer is started, which I find odd...

**Comment by parg** [comment]
I don't

**Comment by parg** [comment]
All files that are required to exist are created on torrent startup so if there is an issue with their creation (permissions or whatever) you get the error when at the computer, not 10 minutes later when you've wandered off.
